### DISCUSSION TITLE
Adding animated object to aircraft

### DISCUSSION THREAD

**Discussion by Degorap** [flybywiresim/aircraft#7104] [Help]
Hello, first of all I want to thank everyone who contributes to this project by creating the best MSFS commercial airliner!!

Now, I ask for help with my project, I am trying to add an animated object to the plane, the way I approach to do it is the following:

1 create an object in Blender
2 export it with one of the several Blender2MSFS exporters out there (no problem here)
3 set it as object in an simple aircraft project (MSFS samples)
4 build the project and get the final files (bin+gltf+textures)
5 merge my object with the originals aircraft objects using the Build.js (with a simplified models.json)

I don't know if this is the right way but it really works, until I add an Armature to my object in Blender to animate it.

After adding the armature (even without creating animations yet), I follow the same steps and everything works fine even until the creation of the aircraft project, I can see in the hangar my animated object like a plane, but then when I merge it with the original aircraft objects using Build.js, the result is a CTD when the simulator is loading.

Could someone give me some guidance on how to fix the problem?

Update: Trying to understand how build.js works (conceptual, I have no knowledge of JS) I find a difference in the object with the armature, it has a "skins" section (gltf file) and I don't find that in the build.js there is a function that processes that section. 

Thank you very much.
D.

**Comment by tracernz** [comment]
The build.js would need to be extended to support this because as you say, skinning (among other things) is not currently supported. As we are building a new model from scratch that won't required this process anymore, we have no immediate plans to work further on this script. We would accept patches if anybody is willing to do that work though.

**Reply by Degorap** [reply]
Thanks for your answer, maybe I'll try it.

A clarification that would help me, making the script consider skinning requires working on the gltf file definition only or is it necessary to work on the BIN file as well? In other words, does the resulting BIN file with the current script have the necessary data?

Thanks you in advance.
D.

**Reply by tracernz** [reply]
That I do not know.
